java/vector/src/main/java/org/apache/arrow/vector/VarBinaryVector.java

```diff
@@ -242,10 +242,12 @@ public void set(int index, NullableVarBinaryHolder holder) {
     assert index >= 0;
     fillHoles(index);
     BitVectorHelper.setValidityBit(validityBuffer, index, holder.isSet);
-    final int dataLength = holder.end - holder.start;
+    final int dataLength = holder.isSet == 0 ? 0 : holder.end - holder.start;
```

Comment:
According to the arrow spec, when a variable width slot is null the length should be zero.  What symptom were you seeing that this is trying to fix?

Reply:
Oh, I see this is on the write path.

I think it is clearer if you handle this the other way:

if (holder.isSet) {
  /// set the bytes
}
lastSet = index;

Reply:
Thanks for the suggestion. 
I have revised the code. Please see if it is what you want.